tests/trainer/test_trainer.py

```diff
@@ -873,148 +873,74 @@ def on_keyboard_interrupt(self, trainer, pl_module):
     assert isinstance(handle_interrupt_callback.exc_info[1], KeyboardInterrupt)
 
 
-def test_gradient_clipping(tmpdir):
-    """
-    Test gradient clipping
-    """
-    tutils.reset_seed()
-
-    model = EvalModelTemplate()
-
-    trainer = Trainer(max_steps=1, max_epochs=1, gradient_clip_val=1.0, default_root_dir=tmpdir)
-
-    old_training_step_and_backward = trainer.fit_loop.epoch_loop.batch_loop.training_step_and_backward
-
-    def training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, hiddens):
-        """
-        wrap the forward step in a closure so second order methods work
-        """
-        # test that gradient is clipped correctly
-        ret_val = old_training_step_and_backward(split_batch, batch_idx, opt_idx, optimizer, hiddens)
-        parameters = model.parameters()
-        grad_norm = torch.norm(torch.stack([torch.norm(p.grad.detach(), 2) for p in parameters]), 2)
-        assert (grad_norm - 1.0).abs() < 0.01, f"Gradient norm != 1.0: {grad_norm}"
-
-        return ret_val
-
-    trainer.fit_loop.epoch_loop.batch_loop.training_step_and_backward = training_step_and_backward
-    # for the test
-    model.prev_called_batch_idx = 0
-
-    trainer.fit(model)
-
-
-def test_gradient_clipping_by_value(tmpdir):
-    """
-    Test gradient clipping by value
-    """
+@pytest.mark.parametrize(
+    "precision,gpus",
+    [pytest.param(32, 0), pytest.param(16, 1, marks=RunIf(min_gpus=1, amp_native=True))],
+)
+def test_gradient_clipping_by_norm(tmpdir, precision, gpus):
+    """Test gradient clipping by norm"""
     tutils.reset_seed()
 
-    model = BoringModel()
-
-    grad_clip_val = 1e-10
+    model = EvalModelTemplate()  # TODO: when precision=16, BoringModel produces NaN, but EvalModelTemplate not
     trainer = Trainer(
+        default_root_dir=tmpdir,
         max_steps=1,
         max_epochs=1,
-        gradient_clip_val=grad_clip_val,
-        gradient_clip_algorithm="value",
-        default_root_dir=tmpdir,
+        gpus=gpus,
```

Comment:
```suggestion
        gpus=int(torch.cuda.is_available()),
```
then we do not need the one argument :]